src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/PublisherEventsPersistor.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.PerformanceAnalyzerApp;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.Action;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.metrics.ExceptionsAndErrors;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.metrics.RcaRuntimeMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence.actions.ActionsSummary;
+import java.time.Instant;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * A listener that persists all actions published by the publisher to rca.sqlite
+ */
+public class PublisherEventsPersistor {
+    public static final String NAME = "publisher_events_persistor";
+    private static final Logger LOG = LogManager.getLogger(PublisherEventsPersistor.class);
+
+    private final Persistable persistable;
+
+    public PublisherEventsPersistor(final Persistable persistable) {
+        this.persistable = persistable;
+    }
+
+    public void persistAction(final Action action) {
+        long timestamp = Instant.now().toEpochMilli();
+        if (persistable != null) {
+            LOG.debug("Action: [{}] published to persistor publisher.", action.name());
+            PerformanceAnalyzerApp.RCA_RUNTIME_METRICS_AGGREGATOR.updateStat(
+                    RcaRuntimeMetrics.ACTIONS_PUBLISHED, this.name(), 1);
```

Comment:
Should we push these lines inside the if block ? If there are no actions, we may not want the counter to the incremented, right ? Do you just wan to track how many times this method is called ?

Reply:
If block is for impactedNodes. This method will be called only when an action is present. We can retail this outside.